### PR TITLE
IGNITE-15223 Control.sh should be able to perform Read Repair on specified partition.

### PR DESCRIPTION
**Pull request by anton-vinogradov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 public class ActivateCommand extends AbstractCommand<Void> {
     /** {@inheritDoc} */
     @Override public void printUsage(Logger logger) {
-        Command.usage(logger, "Activate cluster (deprecated. Use " + SET_STATE.toString() + " instead):", ACTIVATE);
```
Comment (nizhikov) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java:38]: Why this change?

Comment (anton-vinogradov) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java:38]: This chance is to write an explicit [EXPERIMENTAL] flag at UI.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            GridCacheContext<Object, Object> cctx = ignite.context().cache().cache(cacheName).context();
+
+            if (!cctx.gridEvents().isRecordable(EVT_CONSISTENCY_VIOLATION))
+                throw new UnsupportedOperationException("Consistency violation events recording is disabled on cluster.");
+
+            CacheGroupContext grpCtx = cctx.group();
+
+            GridDhtLocalPartition part = grpCtx.topology().localPartition(p);
+
+            if (part != null) {
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:119]: Let's rewrite it with 

```
if (part == null)
    return null;

// Other logic goes here.
```

to reduce indentation level.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+                    ignite.events().localListen(lsnr, EVT_CONSISTENCY_VIOLATION);
+
+                    try {
+                        Set<Object> keys = new HashSet<>();
+
+                        GridCursor<? extends CacheDataRow> cursor =
+                            grpCtx.offheap().dataStore(part).cursor(cctx.cacheId());
+
+                        IgniteCache<Object, Object> cache = ignite.cache(cacheName).withReadRepair();
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:133]: It seems we have to use `withKeepBinary` here to handle case when key class is obscene on the server node. Let's add tests for that case, also(caches with just `BinaryObject` and caches created by `CREATE TABLE`).

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+         */
+        protected VisorConsistencyRepairJob(VisorConsistencyRepairTaskArg arg, boolean debug) {
+            super(arg, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected String run(VisorConsistencyRepairTaskArg arg) throws IgniteException {
+            String cacheName = arg.cacheName();
+            int p = arg.part();
+            int batchSize = 1024;
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:108]: This variable can be inlined

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTaskArg.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ */
+public class VisorConsistencyRepairTaskArg extends IgniteDataTransferObject {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Cache name. */
+    private String cacheName;
+
+    /** Partition. */
+    private Integer part;
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTaskArg.java:37]: Can this variable be `int`?

**Review by nizhikov — APPROVED** [review]
LGTM with the respect of [IGNITE-15482](https://issues.apache.org/jira/browse/IGNITE-15482)

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java (hunk 3 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:133]: Found a bug here.
It Will be fixed at IGNITE-15482

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java (hunk 2 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java (hunk 4 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:108]: Idea is to make it configurable later.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTaskArg.java (hunk 5 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTaskArg.java]: Sure
